src/main/scala/kamon/prometheus/ScrapeDataBuilder.scala

```diff
@@ -140,9 +141,10 @@ class ScrapeDataBuilder(prometheusConfig: PrometheusReporter.Configuration) {
 
 
   private def appendTags(tags: Map[String, String]): Unit = {
-    if(tags.nonEmpty) append("{")
+    val allTags = environmentTags ++ tags
+    if(allTags.nonEmpty) append("{")
```

Comment:
Better performance would be:
```scala
val tagIterator = environmentTags.iterator ++ tags.iterator
val addBraces = tagIterator.hasNext()
if (addBraces) append("{")
// ...
```

Reply:
Iterator `++` much more efficient than Map `++` followed by `.iterator`

Reply:
The results are potentially different though, as concatenating iterators may yield duplicate tags, while concatenating maps ensures that there's only one entry per key.

Exposing multiple labels to Prometheus won't break the Prometheus scraper, but will lead to Prometheus only keeping the one that occurs later in the sequence of labels. 

Personally, I'd opt for having an explicit preference in this exporter instead of relying on Prometheus to do the deduplication. That does bring up the following point: which tag would we prefer when we run into a duplicate? The one from the `Kamon.environment`, or the one from the regular `tags`?

Reply:
In the case above I would say that environment tag should win. Typically this would be used to discern the instance from which data is being collected so all tags for a metric should have the very same tags, it would me a mistake to provide a custom tag that overrides the environment tags.

Thanks for bringing this up, I'll make this makes it to the docs! (see https://github.com/kamon-io/kamon.io/issues/43)